### ISSUE TITLE
[GoPlay] ffmpeg "error reading header" processing multi-period MPD streams

### ISSUE THREAD

**Issue by alard** [yt-dlp/yt-dlp#9262]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Belgium, Netherlands, possibly others

### Provide a description that is worded well enough to be understood

This is related to the recently merged #6654 for the GoPlay extractor, which downloads multi-period MPD streams by concatenating the `fragments` list to create a single stream. This works perfectly fine with some ffmpeg versions, but crashes on others.

I don't know if this is an artifact of the way the fragments are combined (e.g., it repeats the initialization segment for each period) or if there is a clever ffmpeg flag that could be added to make this work on more systems.

*I've had success with:*
* ffmpeg 6.0 on Ubuntu 23.10
* [static ffmpeg 4.1](https://johnvansickle.com/ffmpeg/)
* latest [static ffmpeg for yt-dlp](https://github.com/yt-dlp/FFmpeg-Builds)

*Problems occur with:*
* ffmpeg 5.1.4 on Debian 12
* [static ffmpeg 6.1](https://johnvansickle.com/ffmpeg/)

**More detail about the errors:**

* If it works, the postprocessing step shows a warning about duplicated MOOV atoms. After postprocessing, the cleaned-up output file is playable.

  ffmpeg 6.0 on Ubuntu 23.10:
  ```
  ffprobe S9\ -\ Aflevering\ 1\ \[70dcacc8-726a-4f69-b127-3ce198cb47b0\].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4 
  ffprobe version 6.0-6ubuntu1 Copyright (c) 2007-2023 the FFmpeg developers
    built with gcc 13 (Ubuntu 13.2.0-2ubuntu1)
    configuration: --prefix=/usr --extra-version=6ubuntu1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libvpl --disable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared
    libavutil      58.  2.100 / 58.  2.100
    libavcodec     60.  3.100 / 60.  3.100
    libavformat    60.  3.100 / 60.  3.100
    libavdevice    60.  1.100 / 60.  1.100
    libavfilter     9.  3.100 /  9.  3.100
    libswscale      7.  1.100 /  7.  1.100
    libswresample   4. 10.100 /  4. 10.100
    libpostproc    57.  1.100 / 57.  1.100
  [mov,mp4,m4a,3gp,3g2,mj2 @ 0x55689c0c7940] Found duplicated MOOV Atom. Skipped it
      Last message repeated 1 times
  Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4':
    Metadata:
      major_brand     : mp42
      minor_version   : 1
      compatible_brands: isommp42mp41
    Duration: 01:19:31.52, start: 0.080000, bitrate: 995 kb/s
    Stream #0:0[0x1](und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(progressive), 768x432 [SAR 1:1 DAR 16:9], 992 kb/s, 25 fps, 25 tbr, 25k tbn (default)
      Metadata:
        handler_name    : Bitmovin Video Handler
        vendor_id       : [0][0][0][0]
        encoder         : h264
  ```

* With some other versions, ffmpeg crashes the postprocessing step with warnings "error reading header" and "Invalid data found when processing input":

  ffmpeg 5.1.4 on Debian 12:
  ```
  ffprobe S9\ -\ Aflevering\ 1\ \[70dcacc8-726a-4f69-b127-3ce198cb47b0\].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4
  ffprobe version 5.1.4-0+deb12u1 Copyright (c) 2007-2023 the FFmpeg developers
    built with gcc 12 (Debian 12.2.0-14)
    configuration: --prefix=/usr --extra-version=0+deb12u1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared
    libavutil      57. 28.100 / 57. 28.100
    libavcodec     59. 37.100 / 59. 37.100
    libavformat    59. 27.100 / 59. 27.100
    libavdevice    59.  7.100 / 59.  7.100
    libavfilter     8. 44.100 /  8. 44.100
    libswscale      6.  7.100 /  6.  7.100
    libswresample   4.  7.100 /  4.  7.100
    libpostproc    56.  6.100 / 56.  6.100
  [mov,mp4,m4a,3gp,3g2,mj2 @ 0x5581b58ffec0] error reading header
  S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4: Invalid data found when processing input
  ```

<details>
<summary>Other versions I tried</summary>

* [static ffmpeg 6.1](https://johnvansickle.com/ffmpeg/) does not work:

  ```
  ffmpeg-6.1-amd64-static/ffprobe S9\ -\ Aflevering\ 1\ \[70dcacc8-726a-4f69-b127-3ce198cb47b0\].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4
  ffprobe version 6.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2007-2023 the FFmpeg developers
    built with gcc 8 (Debian 8.3.0-6)
    configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
    libavutil      58. 29.100 / 58. 29.100
    libavcodec     60. 31.102 / 60. 31.102
    libavformat    60. 16.100 / 60. 16.100
    libavdevice    60.  3.100 / 60.  3.100
    libavfilter     9. 12.100 /  9. 12.100
    libswscale      7.  5.100 /  7.  5.100
    libswresample   4. 12.100 /  4. 12.100
    libpostproc    57.  3.100 / 57.  3.100
  [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7924000] error reading header
  S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4: Invalid data found when processing input
  ```

* [static ffmpeg 4.1](https://johnvansickle.com/ffmpeg/) works:

  ```
  ffmpeg-4.4.1-amd64-static/ffprobe S9\ -\ Aflevering\ 1\ \[70dcacc8-726a-4f69-b127-3ce198cb47b0\].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4
  ffprobe version 4.4.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2007-2021 the FFmpeg developers
    built with gcc 8 (Debian 8.3.0-6)
    configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
    libavutil      56. 70.100 / 56. 70.100
    libavcodec     58.134.100 / 58.134.100
    libavformat    58. 76.100 / 58. 76.100
    libavdevice    58. 13.100 / 58. 13.100
    libavfilter     7.110.100 /  7.110.100
    libswscale      5.  9.100 /  5.  9.100
    libswresample   3.  9.100 /  3.  9.100
    libpostproc    55.  9.100 / 55.  9.100
  [mov,mp4,m4a,3gp,3g2,mj2 @ 0x6dd8480] Found duplicated MOOV Atom. Skipped it
      Last message repeated 1 times
  Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4':
    Metadata:
      major_brand     : mp42
      minor_version   : 1
      compatible_brands: isommp42mp41
    Duration: 01:19:31.52, start: 0.080000, bitrate: 995 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 768x432 [SAR 1:1 DAR 16:9], 992 kb/s, 25 fps, 25 tbr, 25k tbn, 50 tbc (default)
      Metadata:
        handler_name    : Bitmovin Video Handler
        vendor_id       : [0][0][0][0]
        encoder         : h264
  ```

* Latest [static ffmpeg for yt-dlp](https://github.com/yt-dlp/FFmpeg-Builds) does work:

  ```
  yt-dlp-ffmpeg/ffmpeg-master-latest-linux64-gpl/bin/ffprobe S9\ -\ Aflevering\ 1\ \[70dcacc8-726a-4f69-b127-3ce198cb47b0\].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4
  ffprobe version N-113753-ga3f69cdec7-20240221 Copyright (c) 2007-2024 the FFmpeg developers
    built with gcc 13.2.0 (crosstool-NG 1.25.0.232_c175b21)
    configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-ffbuild-linux-gnu- --arch=x86_64 --target-os=linux --enable-gpl --enable-version3 --disable-debug --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-openssl --enable-lzma --enable-fontconfig --enable-libharfbuzz --enable-libvorbis --enable-opencl --enable-libpulse --enable-libvmaf --enable-libxcb --enable-xlib --enable-amf --enable-libaom --enable-libaribb24 --enable-avisynth --enable-chromaprint --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libkvazaar --enable-libaribcaption --enable-libass --enable-libbluray --enable-libjxl --enable-libmp3lame --enable-libopus --enable-librist --enable-libssh --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --enable-libvpl --enable-openal --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --disable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --enable-libdrm --enable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags=-pie --extra-libs='-ldl -lgomp' --extra-version=20240221
    libavutil      58. 39.100 / 58. 39.100
    libavcodec     60. 40.100 / 60. 40.100
    libavformat    60. 21.101 / 60. 21.101
    libavdevice    60.  4.100 / 60.  4.100
    libavfilter     9. 17.100 /  9. 17.100
    libswscale      7.  6.100 /  7.  6.100
    libswresample   4. 13.100 /  4. 13.100
    libpostproc    57.  4.100 / 57.  4.100
  [mov,mp4,m4a,3gp,3g2,mj2 @ 0x556cc251dcc0] Found duplicated MOOV Atom. Skipped it
      Last message repeated 1 times
  Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4':
    Metadata:
      major_brand     : mp42
      minor_version   : 1
      compatible_brands: isommp42mp41
    Duration: 01:19:31.52, start: 0.080000, bitrate: 995 kb/s
    Stream #0:0[0x1](und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(progressive), 768x432 [SAR 1:1 DAR 16:9], 992 kb/s, 25 fps, 25 tbr, 25k tbn (default)
        Metadata:
          handler_name    : Bitmovin Video Handler
          vendor_id       : [0][0][0][0]
          encoder         : h264
  ```

</details>

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--username', 'PRIVATE', '--password', 'PRIVATE', 'https://www.goplay.be/video/de-mol/de-mol-s9/de-mol-s9-aflevering-1', '-S', '+size,+br', '--keep-video', '--keep-fragments', '--ffmpeg-location', 'ffmpeg-6.1-amd64-static/ffmpeg']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8 (No ANSI), error utf-8 (No ANSI), screen utf-8 (No ANSI)
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 28e53d60d
[debug] Python 3.11.6 (CPython x86_64 64bit) - Linux-6.5.0-17-generic-x86_64-with-glibc2.38 (OpenSSL 3.0.10 1 Aug 2023, glibc 2.38)
[debug] exe versions: ffmpeg 6.1-static (setts), ffprobe 6.1-static
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2022.12.07, mutagen-1.47.0, requests-2.31.0, sqlite3-3.42.0, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1834 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.12.30 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.12.30 from yt-dlp/yt-dlp)
[GoPlay] Logging in
[GoPlay] Authenticating username
[GoPlay] Authenticating password
[GoPlay] Extracting URL: https://www.goplay.be/video/de-mol/de-mol-s9/de-mol-s9-aflevering-1
[GoPlay] de-mol-s9-aflevering-1: Downloading webpage
[GoPlay] 70dcacc8-726a-4f69-b127-3ce198cb47b0: Downloading JSON metadata
[GoPlay] 70dcacc8-726a-4f69-b127-3ce198cb47b0: Downloading JSON metadata
[GoPlay] 70dcacc8-726a-4f69-b127-3ce198cb47b0: Downloading MPD manifest
[debug] Sort order given by user: +size, +br
[debug] Formats sorted by: hasvid, ie_pref, +size, +br, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 70dcacc8-726a-4f69-b127-3ce198cb47b0: Downloading 1 format(s): f26497bd-f045-4c67-bde3-60dccf5162e3+cd7e7bff-49e6-403a-92ae-96306b41075c
[debug] Invoking dashsegments downloader on "https://dai.google.com/ondemand/dash/content/2615619/vid/10147193212751527/GRQ/streams/a211f2d5-d427-4245-a2cc-f3c96b283208/manifest.mpd"
[dashsegments] Total fragments: 2491
[download] Destination: S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4


removed: downloading fragments


[debug] Invoking dashsegments downloader on "https://dai.google.com/ondemand/dash/content/2615619/vid/10147193212751527/GRQ/streams/a211f2d5-d427-4245-a2cc-f3c96b283208/manifest.mpd"
[dashsegments] Total fragments: 2491
[download] Destination: S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].fcd7e7bff-49e6-403a-92ae-96306b41075c.m4a


removed: downloading fragments


[Merger] Merging formats into "S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].mp4"
[debug] ffmpeg command line: ffmpeg-6.1-amd64-static/ffmpeg -y -loglevel repeat+info -i 'file:S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4' -i 'file:S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].fcd7e7bff-49e6-403a-92ae-96306b41075c.m4a' -c copy -map 0:v:0 -map 1:a:0 -movflags +faststart 'file:S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].temp.mp4'
[debug] ffmpeg version 6.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2023 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      58. 29.100 / 58. 29.100
  libavcodec     60. 31.102 / 60. 31.102
  libavformat    60. 16.100 / 60. 16.100
  libavdevice    60.  3.100 / 60.  3.100
  libavfilter     9. 12.100 /  9. 12.100
  libswscale      7.  5.100 /  7.  5.100
  libswresample   4. 12.100 /  4. 12.100
  libpostproc    57.  3.100 / 57.  3.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x6d329c0] error reading header
[in#0 @ 0x6d32800] Error opening input: Invalid data found when processing input
Error opening input file file:S9 - Aflevering 1 [70dcacc8-726a-4f69-b127-3ce198cb47b0].ff26497bd-f045-4c67-bde3-60dccf5162e3.mp4.
Error opening input files: Invalid data found when processing input

ERROR: Postprocessing: Error opening input files: Invalid data found when processing input
Traceback (most recent call last):
  File "/home/a/src/yt-dlp/yt_dlp/YoutubeDL.py", line 3510, in process_info
    replace_info_dict(self.post_process(dl_filename, info_dict, files_to_move))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/a/src/yt-dlp/yt_dlp/YoutubeDL.py", line 3690, in post_process
    info = self.run_all_pps('post_process', info, additional_pps=info.get('__postprocessors'))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/a/src/yt-dlp/yt_dlp/YoutubeDL.py", line 3672, in run_all_pps
    info = self.run_pp(pp, info)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/a/src/yt-dlp/yt_dlp/YoutubeDL.py", line 3650, in run_pp
    files_to_delete, infodict = pp.run(infodict)
                                ^^^^^^^^^^^^^^^^
  File "/home/a/src/yt-dlp/yt_dlp/postprocessor/common.py", line 23, in run
    ret = func(self, info, *args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/a/src/yt-dlp/yt_dlp/postprocessor/common.py", line 128, in wrapper
    return func(self, info)
           ^^^^^^^^^^^^^^^^
  File "/home/a/src/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 839, in run
    self.run_ffmpeg_multiple_files(info['__files_to_merge'], temp_filename, args)
  File "/home/a/src/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 329, in run_ffmpeg_multiple_files
    return self.real_run_ffmpeg(
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/a/src/yt-dlp/yt_dlp/postprocessor/ffmpeg.py", line 367, in real_run_ffmpeg
    raise FFmpegPostProcessorError(stderr.strip().splitlines()[-1])
yt_dlp.postprocessor.ffmpeg.FFmpegPostProcessorError: Error opening input files: Invalid data found when processing input
```

**Comment by bashonly** [comment]
Looks like the same issue as #8641

It's a regression in ffmpeg 6.1 (and backported to 5.1.4 apparently), which was fixed in 6.1.1

**Comment by alard** [comment]
Created a ticket for ffmpeg to see if they can backport the 6.1.1 fix to 5.1.x: https://trac.ffmpeg.org/ticket/10879
